### ISSUE TITLE
ksh93 shared-state command substitution syntax isn't handled correctly

### ISSUE THREAD

**Issue by JohnoKing** [vim/vim#9514]
### Steps to reproduce

The following ksh93 script uses shared-state command substitutions to print a few lines. The syntax is valid but vim doesn't seem to handle the script correctly:
```ksh
#!/bin/ksh

print $(echo 'string one')
print ${ echo 'string two'; }
print ${ echo 'string three' ;}
print ${ echo 'string four' }
exit 0
```

### Expected behaviour

KornShell shared-state command substitutions should be rendered the same way normal command substitutions are.

### Version of Vim

VIM - Vi IMproved 8.2 (2019 Dec 12, compiled Dec 25 2021 11:17:40) Included patches: 1-3890  (using sh.vim from current git commit)

### Environment

Operating system: Arch Linux x86_64
Terminal: kitty 0.24.1
$TERM: xterm-kitty
Shell: ksh93u+m

**Comment by chrisbra** [comment]
is that fixed by your #9513 ?

**Comment by chrisbra** [comment]
cc @cecamp 

For now, you can define `:g:no_sh_error` (but this may also hide other errors)
